coap-gateway/service/signIn.go

```diff
@@ -138,7 +139,20 @@ func signInPostHandler(req *mux.Message, client *Client, signIn CoapSignInReq) {
 	}
 
 	if newDevice {
-		deviceSubscriber, err := grpcgwClient.NewDeviceSubscriber(req.Context, signIn.DeviceID, client.server.rdClient, client.server.resourceSubscriber)
+		deviceSubscriber, err := grpcgwClient.NewDeviceSubscriber(req.Context, signIn.DeviceID, func() func() (when time.Time, err error) {
+			var count uint64
+			maxRand := client.server.config.APIs.COAP.KeepAlive.Timeout / 2
```

Comment:
We wanted to handle this somewhere else, not to have everywhere / 2 where this property is used. Nope?

Reply:
no this is used to calculate next time of reconnet : Timeout + rand(Timeout/2) - we want to avoid reconnect all client in same time.